crates/editor/src/editor.rs

```diff
@@ -18208,10 +18242,16 @@ impl Editor {
         if self.buffer().read(cx).is_singleton() || self.is_buffer_folded(buffer_id, cx) {
             return;
         }
+
         let folded_excerpts = self.buffer().read(cx).excerpts_for_buffer(buffer_id, cx);
         self.display_map.update(cx, |display_map, cx| {
             display_map.fold_buffers([buffer_id], cx)
         });
+
+        self.selections.change_with(cx, |selections| {
+            selections.remove_selections_from_buffer(buffer_id);
```

Comment:
I’m a bit skeptical of this change. I don’t think folding should clear selections inside it. You might fold buffers for various reasons, but in the end it’s just a view and shouldn’t be a destructive action.

That said, I think `editor::SelectAllMatches` shouldn’t select matches in folded buffers. If you are up for making that change.